### PR TITLE
Feature | V1-235 - Added badges for Pending request and Assessments MenuLinks for Manager

### PR DESCRIPTION
**Pull request by ostapdans99**
- Frontend

1. Added Badges for Pending request and Assessments MenuLinks for Manager
2. Refactored Layout folder

For reviewers: No changes were made for the Search logic and files.

- Backend

1. Added `pendingRequestsAmount` and `pendingAssessmentsAmount` calculated fields to profile response.

### CODE REVIEW

File: frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          handleConfirmLogOutOpen={handleConfirmLogOutOpen}
+          toggleMobileMenu={toggleMobileMenu}
+        />
+      </GridHeader>
+      <>
+        <GridMenu isSqueeze={isSqueeze}>
+          <Menu isSqueeze={isSqueeze} toggleSqueeze={toggleSqueeze} />
+        </GridMenu>
+        <PageWrapper isSqueeze={isSqueeze}>
+          <Suspense fallback={<Loader color="primary" />}>{children}</Suspense>
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayout.tsx:43]: Should we add Loader type here?

File: frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayoutContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { FC, useState } from 'react';
+
+import { useLogOut } from 'api/logOut/';
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayoutContainer.tsx:3]:
```suggestion
import { useLogOut } from 'api/logOut';
```

File: frontend/src/components/Layout/AuthorizedLayout/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      <HeaderContent>
+        <SearchCourses />
+        <Notifications
+          notifications={notifications}
+          isMobileMenuOpen={isMobileMenuOpen}
+          toggleMobileMenu={toggleMobileMenu}
+        />
+        <UserBlock to={PATHS.profile}>
+          <Avatar avatar={avatar} size={Size.small} />
+          <UserName>{`${firstName} ${lastName}`}</UserName>
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/AuthorizedLayout/Header/Header.tsx:46]: maybe we should also implement a helper like on the BE side?
`export const convertToFullName = (firstName: string, lastName: string) => `${firstName} ${lastName}`;
`

File: frontend/src/components/Layout/AuthorizedLayout/Header/SearchCourses/SearchCoursesContainer.tsx [hunk 4]
```diff
@@ -1,13 +1,14 @@
 import React, { useState, useEffect } from 'react';
 import { useSnackbar } from 'notistack';
 
-import { ICourse } from 'types/course';
-import SearchCourses from 'components/Layout/Header/SearchCourses/SearchCourses';
+import useSearchAllCourses from 'api/courses/searchAllCourses';
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/AuthorizedLayout/Header/SearchCourses/SearchCoursesContainer.tsx:4]:
```suggestion
import { useSearchAllCourses } from 'api/courses';
```

File: frontend/src/components/Layout/AuthorizedLayout/Menu/Badge/utils.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { BadgeType } from 'enums/badgeType';
+
+import { getUserProfileCache } from 'utils/cache';
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/AuthorizedLayout/Menu/Badge/utils.ts:3]:
```suggestion
import { BadgeType } from 'enums/badgeType';
import { getUserProfileCache } from 'utils/cache';
```

File: frontend/src/components/Layout/AuthorizedLayout/MobileMenu/MobileMenuContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+const MobileMenuContainer: FC<TMenuMobileContainerProps> = ({ isMobileMenuOpen, ...props }) => {
+  const windowSize = useGetWindowSizeLabel();
+
+  const isMobileWindowSize = [WINDOW_SIZE.sm.name, WINDOW_SIZE.xs.name].includes(windowSize);
+  const isShowMenu = isMobileMenuOpen && isMobileWindowSize;
+
+  const handleSpaceHolderClick = (event: BaseSyntheticEvent) => {
+    event.stopPropagation();
+  };
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/AuthorizedLayout/MobileMenu/MobileMenuContainer.tsx:17]: can we destructure the event at once, as we discussed?

File: frontend/src/components/Layout/AuthorizedLayout/MobileMenu/styled.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+});
+
+export const SpaceHolder = styled('div')({
+  flex: '1 1 0',
+});
+
+export const MobileUserBlock = styled('div')({
+  display: 'inline-flex',
+  justifyContent: 'space-between',
+  padding: '0px 18px 0px 12px',
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/AuthorizedLayout/MobileMenu/styled.ts:38]:
```suggestion
  padding: '0 18px 0 12px',
```

File: frontend/src/components/Layout/DefaultLayout/DefaultLayout.tsx [hunk 8]
```diff
@@ -1,15 +1,11 @@
-import React, { Suspense } from 'react';
-import Grid from '@mui/material/Grid';
+import { FC, Suspense } from 'react';
 import { Helmet } from 'react-helmet';
+import Grid from '@mui/material/Grid';
```
[the remark this answers is not in the document]
Comment (ostapdans99) [reply frontend/src/components/Layout/DefaultLayout/DefaultLayout.tsx:3]: done

[the remark this answers is not in the document]
Comment (ostapdans99) [reply frontend/src/components/Layout/DefaultLayout/DefaultLayout.tsx:3]: done

File: frontend/src/components/Layout/AuthorizedLayout/MobileMenu/styled.ts (hunk 7 above)
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/MobileMenu/styled.ts:38]: done

File: frontend/src/components/Layout/AuthorizedLayout/MobileMenu/styled.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+export const MobileMenuSlide = styled('div')({
+  position: 'absolute',
+  top: HEADER_HEIGHT_IPAD,
+  right: 0,
+  display: 'flex',
+  flexDirection: 'column',
+  width: `${MOBILE_MENU_WIDTH}`,
+  height: `calc(100% - ${HEADER_HEIGHT_IPAD})`,
+  padding: '8px 0px 40px 0px',
```
[the remark this answers is not in the document]
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/MobileMenu/styled.ts:18]: done

File: frontend/src/components/Layout/AuthorizedLayout/MobileMenu/MobileMenuContainer.tsx (hunk 6 above)
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/MobileMenu/MobileMenuContainer.tsx:17]: done

File: frontend/src/components/Layout/AuthorizedLayout/MobileMenu/MobileMenu.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
+import { FC } from 'react';
+
+import { Slide } from '@mui/material';
```
[the remark this answers is not in the document]
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/MobileMenu/MobileMenu.tsx:3]: done

File: frontend/src/components/Layout/AuthorizedLayout/Menu/Badge/utils.ts (hunk 5 above)
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/Menu/Badge/utils.ts:3]: done

File: frontend/src/components/Layout/AuthorizedLayout/Header/SearchCourses/SearchCoursesContainer.tsx (hunk 4 above)
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/Header/SearchCourses/SearchCoursesContainer.tsx:4]: done

File: frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayoutContainer.tsx (hunk 2 above)
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayoutContainer.tsx:3]: done

File: frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayout.tsx (hunk 1 above)
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayout.tsx:43]: done

File: frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayout.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
+import { FC, Suspense } from 'react';
+import Grid from '@mui/material/Grid';
```
[the remark this answers is not in the document]
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/AuthorizedLayout.tsx:2]: done

File: frontend/src/components/CustomNavLink/CustomNavLink.tsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+      ref={ref}
+      className={({ isActive }) =>
+        ` ${defaultClassName} ${isActive ? activeClassName : inactiveClassName}`
+      }
+      {...props}
+    >
+      {children}
+    </NavLink>
+  );
+};
```
[the remark this answers is not in the document]
Comment (ostapdans99) [reply frontend/src/components/CustomNavLink/CustomNavLink.tsx:21]: done

File: frontend/src/components/Layout/AuthorizedLayout/Header/Header.tsx (hunk 3 above)
Comment (ostapdans99) [reply frontend/src/components/Layout/AuthorizedLayout/Header/Header.tsx]: done

**Review by irina-sinitsyna — APPROVED** [review]

**Pull request merged**